packages/grafana-data/src/transformations/transformers/transpose.ts

```diff
@@ -97,17 +100,17 @@ function determineFieldType(fieldTypes: FieldType[]): FieldType {
   return uniqueFieldTypes.size === 1 ? [...uniqueFieldTypes][0] : FieldType.string;
 }
 
-function fieldValuesAsStrings(field: Field, values: unknown[]) {
+function fieldValuesAsStrings(field: Field, values: unknown[], emptyValue: SpecialValue) {
   switch (field.type) {
     case FieldType.time:
     case FieldType.number:
     case FieldType.boolean:
     case FieldType.string:
-      return values.map((v) => `${v}`);
+      return values.map((v) => (v !== undefined && v !== null ? `${v}` : getSpecialValue(emptyValue)));
     case FieldType.enum:
       // @ts-ignore
-      return values.map((v) => field.config.type!.enum!.text![v]);
+      return values.map((v) => field.config.type!.enum!.text![v] ?? getSpecialValue(emptyValue));
     default:
-      return values.map((v) => JSON.stringify(v));
+      return values.map((v) => (v !== undefined && v !== null ? JSON.stringify(v) : getSpecialValue(emptyValue)));
```

Comment:
i know people hate me for using `v != null`, but it's measurable faster when looping over thousands of values, and is shorter to read.